tests/FakeItEasy.Specs/SelfInitializedFakesSpecs.cs

```diff
@@ -67,28 +209,181 @@ public static class SelfInitializedFakesSpecs
+            // defined by which methods are called, and is completely inflexible with
+            // regard to the order of calls.
+            "Then the playback fake throws a recording exception"
+                .x(() => exception.Should().BeAnExceptionOfType<RecordingException>());
+        }
+
+        [Scenario]
+        public static void SelfInitializingExceptionWhileRecording(
+            InMemoryStorage inMemoryStorage,
+            ILibraryService realServiceWhileRecording,
+            ILibraryService realServiceDuringPlayback,
+            Exception originalException,
+            Exception recordingException,
+            Exception playbackException)
+        {
+            "Given a call storage object"
+                .x(() => inMemoryStorage = new InMemoryStorage());
+
+            "And an exception"
+                .x(() => originalException = new InvalidOperationException());
+
+            "And a real service to wrap while recording"
+                .x(() => realServiceWhileRecording = A.Fake<ILibraryService>());
+
+            "And the real service throws the exception when getting the title for book 1"
+                .x(() =>
+                {
+                    realServiceWhileRecording = A.Fake<ILibraryService>();
+                    A.CallTo(() => realServiceWhileRecording.GetTitle("1")).Throws(originalException);
+                });
+
+            "And a real service to wrap while playing back"
+                .x(() => realServiceDuringPlayback = A.Fake<ILibraryService>());
+
+            "When I use a self-initialized fake in recording mode to get the title for book 1"
+                .x(() =>
+                {
+                    using (var recorder = new RecordingManager(inMemoryStorage))
+                    {
+                        var fakeService = A.Fake<ILibraryService>(options => options
+                            .Wrapping(realServiceWhileRecording).RecordedBy(recorder));
+
+                        recordingException = Record.Exception(() => fakeService.GetTitle("1"));
+                    }
+                });
+
+            "And I use a self-initialized fake in playback mode to get the title for book 1"
+                .x(() =>
+                {
+                    using (var recorder = new RecordingManager(inMemoryStorage))
+                    {
+                        var fakeService = A.Fake<ILibraryService>(options => options
+                            .Wrapping(realServiceDuringPlayback).RecordedBy(recorder));
+
+                        playbackException = Record.Exception(() => fakeService.GetTitle("1"));
+                    }
+                });
+
+            "Then the recording fake throws the original exception"
+                .x(() => recordingException.Should().BeSameAs(originalException));
+
+            "But the playback fake throws a recording exception"
+                .x(() => playbackException.Should().BeAnExceptionOfType<RecordingException>());

```

Comment:
The wording is a little confusing. `recordingException` is the exception thrown while recording, but `playbackException` is a `RecordingException`... Renaming the exceptions to `exceptionWhileRecording` and `exceptionWhilePlayingBack` (or anything else) would probably help.